### PR TITLE
cpprestsdk: add components + bump boost version + remove short_paths

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **cpprestsdk/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
well RIP #2348

File: recipes/cpprestsdk/all/conanfile.py [hunk 1]
```diff
@@ -109,37 +107,40 @@ def package(self):
...
+        self.cpp_info.components["cpprest"].libs = tools.collect_libs(self)
+        self.cpp_info.components["cpprest"].requires = ["cpprestsdk_boost_internal", "cpprestsdk_openssl_internal"]
         if self.settings.os == "Linux":
-            self.cpp_info.system_libs.append("pthread")
+            self.cpp_info.components["cpprest"].system_libs.append("pthread")
         elif self.settings.os == "Windows":
             if self.options.get_safe("http_client_impl") == "winhttp":
-                self.cpp_info.system_libs.append("winhttp")
+                self.cpp_info.components["cpprest"].system_libs.append("winhttp")
             if self.options.get_safe("http_listener_impl") == "httpsys":
```
Comment (prince-chrismc) [inline recipes/cpprestsdk/all/conanfile.py:124]: should there be a component for https://github.com/microsoft/cpprestsdk/blob/319dd31f7ba8752683aaab2679a1c8fefcf48ef3/Release/src/CMakeLists.txt#L269?

Comment (SpaceIm) [reply recipes/cpprestsdk/all/conanfile.py:124]: good catch, I'll fix that

**Comment by SpaceIm** [comment]
Technically no, but it's how imported targets are defined in official config files. Less surprise for consumers.

**Comment by prince-chrismc** [comment]
FYI @garethsb-sony this PR will affect you guys :wave:

**Comment by SpaceIm** [comment]
It shouldn't break downstream projects using (specific conan) `cpprestsdk::cpprestsdk` imported target. Though for transparent integration `cpprestsdk::cpprest` imported target should be used.

File: recipes/cpprestsdk/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/cpprestsdk/all/conanfile.py:124]: Well, after digging into CMakeLists, this component can't exist in current recipe. `http_client_impl` should be updated to add 
`winhttppal` value: https://github.com/microsoft/cpprestsdk/blob/319dd31f7ba8752683aaab2679a1c8fefcf48ef3/Release/src/CMakeLists.txt#L134
Maybe in another PR?

**Pull request merged**